packages/framework/tree-agent/src/typeGeneration.ts

```diff
@@ -87,41 +110,50 @@ export function generateEditTypesForPrompt(
  * @remarks The return type of this function is designed to work with Typechat's createZodJsonValidator as well as be used as the JSON schema for OpenAi's structured output response format.
  */
 function generateEditTypes(
-	schema: SimpleTreeSchema,
+	schemas: Iterable<SimpleTreeSchema>,
 	objectCache: MapGetSet<SimpleNodeSchema, ZodTypeAny>,
 	bindableSchemas: Map<string, BindableSchema>,
 ): {
 	domainTypes: Record<string, ZodTypeAny>;
 } {
 	const domainTypes: Record<string, ZodTypeAny> = {};
-	for (const name of schema.definitions.keys()) {
-		domainTypes[name] = getOrCreateType(
-			schema.definitions,
-			name,
-			objectCache,
-			bindableSchemas,
-		);
+	for (const schema of schemas) {
+		for (const name of schema.definitions.keys()) {
+			domainTypes[name] = getOrCreateType(
```

Comment:
Can't this end up overwriting keys in `domainTypes` if several schemas share... whatever is in `schema.definitions`? Doesn't sound like it's the property keys of each schema but not sure.

Reply:
Yup, it can, but the getOrCreateType ensures that it will only ever be overwritten with an identical value. I'll add a comment.